demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                        unsigned predictedClass = results[FLAGS_nt*i+j];
+                        if (predictedClass == rightClasses[i]) {
+                            isPredictionRight = true;
+                            predictedRight = labels[predictedClass];
+                            break;
+                        }
+                    }
+                    if (isPredictionRight) {
+                        predictedLabels.push_back(predictedRight);
+                        rightPredictionsCount++;
+                    }
+                    else {
+                        predictedLabels.push_back(firstPredicted);
+                    }
+                    totalPredictionsCount++;
+
+                    shownImagesInfo.push_back(
+                        std::make_tuple(completedIRInfo.images[i].mat, predictedLabels[i], isPredictionRight));
+                }
+
+                framesNum += FLAGS_b;
+                gridMat.listUpdate(shownImagesInfo);
+
+                avgFPS = 1. / (((cv::getTickCount() - startTickCount) / cv::getTickFrequency()) / framesNum);
+                
+                std::vector<long long> processingEndTimes;
+                cv::Mat showMat = gridMat.getMat(processingEndTimes, hasClasses);
+                if (!FLAGS_no_show && FLAGS_delay != -1) {
+                    cv::imshow("main", showMat);
+                    key = static_cast<char>(cv::waitKey(delay));
+                }
+
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    latencySum += ((processingEndTimes[i] - completedIRInfo.images[i].startTime)
+                                   / cv::getTickFrequency());
+                }
+                avgLatency = latencySum / framesNum;
+                accuracy = static_cast<double>(rightPredictionsCount) / totalPredictionsCount;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, hasClasses, presenter);
+            }
+            else if (!emptyInferRequests.empty()) {
+                mutex.unlock();
+                cv::Mat nextImage = inputImages[nextImageIndex];
+                long long IRstartTime = cv::getTickCount();
+                resizeImage(nextImage, modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                    IRInfo::IRImage(nextImage,
+                                    std::stoul(hasClasses ? classIndexes[nextImageIndex] : "0"),
+                                    IRstartTime));
+                nextImageIndex++;
+                if (nextImageIndex == inputImagesCount) {
+                    nextImageIndex = 0;
+                }
+                if (emptyInferRequests.front().images.size() == FLAGS_b) {
+                    auto emptyIR = emptyInferRequests.front();
+                    auto& ir = emptyIR.ir;
+                    auto irImages = emptyIR.images;
+
+                    emptyIR.ir.SetCompletionCallback([&ir,
```

Comment:
This stores a reference to `ir` which will be destroyed before the callback triggering as well as `emptyIR`.